From f53ef1f5fef3351dc1836c782a681b9b06e9d171 Mon Sep 17 00:00:00 2001
From: Linus Torvalds <torvalds@linux-foundation.org>
Date: Wed, 29 Nov 2017 10:30:13 -0800
Subject: [PATCH 03/10] kallsyms: take advantage of the new '%px' format

commit 668533dc0764b30c9dd2baf3ca800156f688326b upstream

The conditional kallsym hex printing used a special fixed-width '%lx'
output (KALLSYM_FMT) in preparation for the hashing of %p, but that
series ended up adding a %px specifier to help with the conversions.

Use it, and avoid the "print pointer as an unsigned long" code.

Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 kernel/kallsyms.c | 4 ++--
 kernel/module.c   | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/kallsyms.c b/kernel/kallsyms.c
index 6a3b249a2ae1..45ef20016b76 100644
--- a/kernel/kallsyms.c
+++ b/kernel/kallsyms.c
@@ -603,10 +603,10 @@ static int s_show(struct seq_file *m, void *p)
 		 */
 		type = iter->exported ? toupper(iter->type) :
 					tolower(iter->type);
-		seq_printf(m, "%pK %c %s\t[%s]\n", (void *)iter->value,
+		seq_printf(m, "%px %c %s\t[%s]\n", (void *)iter->value,
 			   type, iter->name, iter->module_name);
 	} else
-		seq_printf(m, "%pK %c %s\n", (void *)iter->value,
+		seq_printf(m, "%px %c %s\n", (void *)iter->value,
 			   iter->type, iter->name);
 	return 0;
 }
diff --git a/kernel/module.c b/kernel/module.c
index 0cec78bed165..74cf4f7c419d 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -4155,7 +4155,7 @@ static int m_show(struct seq_file *m, void *p)
 		   mod->state == MODULE_STATE_COMING ? "Loading" :
 		   "Live");
 	/* Used by oprofile and other similar tools. */
-	seq_printf(m, " 0x%pK", mod->core_layout.base);
+	seq_printf(m, " 0x%px", mod->core_layout.base);
 
 	/* Taints info */
 	if (mod->taints)
-- 
2.17.1

